### PR TITLE
fs: Correct impl of `ZenFS::DeleteFile` when target does not exist

### PR DESCRIPTION
**Pull request by metaspace**
Return error when trying to delete non existing file, to align with
`PosixFileSystem::DeleteFile()`:

```c++
    if (unlink(fname.c_str()) != 0) {
      result = IOError("while unlink() file", fname, errno);
    }
```

### CODE REVIEW

File: fs/fs_zenfs.cc [hunk 1]
```diff
@@ -417,6 +417,8 @@ IOStatus ZenFS::DeleteFile(std::string fname) {
     } else {
       zoneFile.reset();
     }
+  } else {
+    s = IOStatus::NotFound("ZenfFS::DeleteFile(): File not found");
```
Comment (aravind-wdc) [inline fs/fs_zenfs.cc:412]: Typo: ZenfFS -> ZenFS.
Could we log the filename which was not found, will be useful in debugging issues.

Comment (metaspace) [reply fs/fs_zenfs.cc]: Thanks @aravind-wdc. Pushed a fix.

**Review by yhr — APPROVED** [review]
Looks good to me

**Comment by yhr** [comment]
Please submit separate PRs for fixups (like the locking patch in this pr) and features / bugfixes next time.

**Pull request merged**